### PR TITLE
LPS-124108 Add endpoints to delete, get and upsert documents by external reference code

### PR DESCRIPTION
**Pull request by javierdearcos**
The issue with deprecating older methods is that we force backend users to learn the _externalReferenceCode_ pattern when just calling the methods without it is enough (and it will be filled automatically).

### CODE REVIEW

File: source-formatter-api-signature-check-exceptions.txt [hunk 1]
```diff
@@ -224,6 +224,8 @@ com.liferay.portal.kernel.repository.capabilities.TrashCapability#moveFileShortc
 com.liferay.portal.kernel.repository.capabilities.TrashCapability#moveFolderFromTrash(long,Folder,Folder,ServiceContext)
 com.liferay.portal.kernel.repository.DocumentRepository#addFileEntry(long,long,String,String,String,String,String,File,ServiceContext)
 com.liferay.portal.kernel.repository.DocumentRepository#addFileEntry(long,long,String,String,String,String,String,InputStream,long,ServiceContext)
+com.liferay.portal.kernel.repository.DocumentRepository#addFileEntry(String,long,long,String,String,String,String,String,File,ServiceContext)
```
Comment (nhpatt) [inline source-formatter-api-signature-check-exceptions.txt:227]: lol

File: modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml [hunk 2]
```diff
@@ -9278,6 +9283,98 @@ paths:
...
+                            schema:
+                                $ref: "#/components/schemas/Document"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/Document"
+                    description:
+                        ""
+            tags: [ "Document" ]
+        put:
+            description:
```
Comment (nhpatt) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml:9340]: missing @review but nice comment :D

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentResourceImpl.java [hunk 3]
```diff
@@ -224,6 +236,17 @@ public EntityModel getEntityModel(MultivaluedMap multivaluedMap) {
 				_expandoTableLocalService));
 	}
 
+	@Override
+	public Document getSiteDocumentByExternalReferenceCode(
+			String externalReferenceCode, Long siteId)
+		throws Exception {
+
+		FileEntry fileEntry = _dlAppService.getFileEntryByExternalReferenceCode(
```
Comment (nhpatt) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentResourceImpl.java:244]: I think Brian prefers this inlined.

**Review by nhpatt — APPROVED** [review]

**Comment by javierdearcos** [comment]
Just force pushed with the things you have mentioned

**Comment by nhpatt** [comment]
ci:forward

**Comment by nhpatt** [comment]
ci:forward

**Comment by nhpatt** [comment]
ci:forward

**Comment by nhpatt** [comment]
ci:forward

**Comment by nhpatt** [comment]
ci:forward

**Comment by nhpatt** [comment]
ci:stop

**Comment by nhpatt** [comment]
ci:forward

**Comment by nhpatt** [comment]
ci:forward

**Comment by nhpatt** [comment]
Missing SF and `ant build-service-portlet-documentlibrary` in portal-impl

**Pull request closed without merging**